tools/image-tag

```diff
@@ -12,7 +12,7 @@ if [ -n "$IMAGE_TAG" ]; then
   exit 0
 fi
 
-WORKING_SUFFIX=$(if git status --porcelain | grep -qE '^(?:[^?][^ ]|[^ ][^?])\s'; then echo "-WIP"; else echo ""; fi)
+WORKING_SUFFIX=$(if git status --porcelain | grep -qE '^[^ ][^?]\s'; then echo "-WIP"; else echo ""; fi)
```

Comment:
The current regex won't match on my `git status --procelain` output.
```bash
$ git status --porcelain
 M crd.mk
 D tests.mk
?? new_file
```
So we still need some kind of grouping with or. The old regex didn't match on ?? too. What was the original intention of this regex?

Reply:
I think the goal was to add "-WIP" either when
1. there are local changes (`git diff` shows something), or
2. if there is changes in the index (`git diff --cached` shows something).

Your patch only captures the 2nd case. What about the following instead?

```
if git status --porcelain | grep -qE '^([^?][^ ]|[^ ][^?])\s'; then echo "-WIP"; else echo ""; fi
```

Reply:
In that case, what about removing `$WORKING_SUFFIX` completely? It is weird to me to add -WIP only for changes in the index.